pkg/controller/jenkins/configuration/base/reconcile.go

```diff
@@ -173,24 +183,30 @@ func (r *ReconcileJenkinsBaseConfiguration) ensureResourcesRequiredForJenkinsPod
 	}
 	r.logger.V(log.VDebug).Info("ConfigurationAsCode Secret and ConfigMap added watched labels")
 
-	if err := r.createRBAC(metaObject); err != nil {
+	httpServiceName := resources.GetJenkinsHTTPServiceName(r.Configuration.Jenkins)
+	if err := r.createService(metaObject, httpServiceName, r.Configuration.Jenkins.Spec.Service); err != nil {
 		return err
 	}
-	r.logger.V(log.VDebug).Info("Service account, role and role binding are present")
+	r.logger.V(log.VDebug).Info("Jenkins HTTP Service is present")
 
-	if err := r.ensureExtraRBAC(metaObject); err != nil {
+	if err := r.createRoute(metaObject, httpServiceName); err != nil && routeAPIFound {
```

Comment:
if verifyRouteApi() && err := r.createRoute(metaObject, httpServiceName); err != nil

Reply:
On every reconcilation we will be checking if the route api is available in this case